File: android/expoview/src/main/java/host/exp/exponent/notifications/schedulers/SchedulerModel.java
package host.exp.exponent.notifications.schedulers;

import java.util.HashMap;

import host.exp.exponent.kernel.ExperienceKey;

public interface SchedulerModel {

  long getNextAppearanceTime();

  boolean shouldBeTriggeredByAction(String action);

  ExperienceKey getOwnerExperienceKey();

  int getNotificationId();

  String getIdAsString();

  HashMap<String, Object> getDetails();

  boolean canBeRescheduled();

  String saveAndGetId();

  void remove();

}

File: android/expoview/src/main/java/host/exp/exponent/notifications/schedulers/SchedulerModel.kt
package host.exp.exponent.notifications.schedulers

import host.exp.exponent.kernel.ExperienceKey
import java.util.*

interface SchedulerModel {
  val nextAppearanceTime: Long
  fun shouldBeTriggeredByAction(action: String?): Boolean
  val ownerExperienceKey: ExperienceKey
  val notificationId: Int
  val idAsString: String
  fun getDetailsMap(): HashMap<String, Any>?
  fun setDetailsFromMap(detailsMap: HashMap<String, Any>)
  fun canBeRescheduled(): Boolean
  fun saveAndGetId(): String
  fun remove()
}
